domain/repositories/champion.go

```diff
@@ -0,0 +1,121 @@
+package repositories
+
+import (
+	"database/sql"
+	"errors"
+
+	"github.com/wizelineacademy/golang-bootcamp-2020/domain/models"
+)
+
+// Errors
+
+// ErrNoRecord is a custom error used in case no Champion is found in the database
+var ErrNoRecord = errors.New("repositories: no matching record found")
+
+// ChampionRepository defines the interface used by a Champion struct to access its repositories methods
+type ChampionRepository interface {
+	GetSingle(id int) (*models.Champion, error)
+	GetMultiple(limit int) ([]*models.Champion, error)
+	Insert(champion *models.Champion) (int, error)
+}
+
+// ChampRepo defines the link between the Champion and the Database
+type ChampRepo struct {
+	DB *sql.DB
+}
+
+// NewChampRepo returns an inituialized ChampRepo struct
+func NewChampRepo(DB *sql.DB) *ChampRepo {
+	return &ChampRepo{DB}
+}
+
+// GetSingle gets a single database row and returns it as a Champion
+func (cr *ChampRepo) GetSingle(id int) (*models.Champion, error) {
+	// SQL statement
+	stmt := `SELECT  name, lore, created FROM champions WHERE id = ?`
```

Comment:
This should be on the Repository, and as it won't change, you should declare it as a constant.

This applies to the other statement on the other function.

As you may see, you are creating it here, and just passing it to the next layer. Why do that if you could simply use it right on the repository layer?